milli/src/update/index_documents/helpers/grenad_helpers.rs

```diff
@@ -0,0 +1,276 @@
+    builder.compression_type(typ);
+    if let Some(level) = level {
+        builder.compression_level(level);
+    }
+    builder.build(file)
+}
+
+pub fn create_sorter(
+    merge: MergeFn,
+    chunk_compression_type: grenad::CompressionType,
+    chunk_compression_level: Option<u32>,
+    max_nb_chunks: Option<usize>,
+    max_memory: Option<usize>,
+) -> grenad::Sorter<MergeFn> {
+    let mut builder = grenad::Sorter::builder(merge);
+    builder.chunk_compression_type(chunk_compression_type);
+    if let Some(level) = chunk_compression_level {
+        builder.chunk_compression_level(level);
+    }
+    if let Some(nb_chunks) = max_nb_chunks {
+        builder.max_nb_chunks(nb_chunks);
+    }
+    if let Some(memory) = max_memory {
+        builder.dump_threshold(memory);
+        builder.allow_realloc(false);
+    }
+    builder.build()
+}
+
+pub fn sorter_into_reader(
+    sorter: grenad::Sorter<MergeFn>,
+    indexer: GrenadParameters,
+) -> Result<grenad::Reader<File>> {
+    let mut writer = tempfile::tempfile().and_then(|file| {
+        create_writer(indexer.chunk_compression_type, indexer.chunk_compression_level, file)
+    })?;
+    sorter.write_into(&mut writer)?;
+    Ok(writer_into_reader(writer)?)
+}
+
+pub fn writer_into_reader(writer: grenad::Writer<File>) -> Result<grenad::Reader<File>> {
+    let mut file = writer.into_inner()?;
+    file.seek(SeekFrom::Start(0))?;
+    grenad::Reader::new(file).map_err(Into::into)
+}
+
+pub unsafe fn into_clonable_grenad(
+    reader: grenad::Reader<File>,
+) -> Result<grenad::Reader<CursorClonableMmap>> {
+    let file = reader.into_inner();
+    let mmap = memmap::Mmap::map(&file)?;
+    let cursor = io::Cursor::new(ClonableMmap::from(mmap));
+    let reader = grenad::Reader::new(cursor)?;
+    Ok(reader)
+}
+
+pub fn merge_readers<R: io::Read>(
+    readers: Vec<grenad::Reader<R>>,
+    merge_fn: MergeFn,
+    indexer: GrenadParameters,
+) -> Result<grenad::Reader<File>> {
+    let mut merger_builder = grenad::MergerBuilder::new(merge_fn);
+    merger_builder.extend(readers);
+    let merger = merger_builder.build();
+    let mut writer = tempfile::tempfile().and_then(|file| {
+        create_writer(indexer.chunk_compression_type, indexer.chunk_compression_level, file)
+    })?;
+    merger.write_into(&mut writer)?;
+    let reader = writer_into_reader(writer)?;
+    Ok(reader)
+}
+
+#[derive(Debug, Clone, Copy)]
+pub struct GrenadParameters {
+    pub chunk_compression_type: CompressionType,
+    pub chunk_compression_level: Option<u32>,
+    pub max_memory: Option<usize>,
+    pub max_nb_chunks: Option<usize>,
+}
+
+impl Default for GrenadParameters {
+    fn default() -> Self {
+        Self {
+            chunk_compression_type: CompressionType::None,
+            chunk_compression_level: None,
+            max_memory: None,
+            max_nb_chunks: None,
+        }
+    }
+}
+
+impl GrenadParameters {
+    pub fn max_memory_by_thread(&self) -> Option<usize> {
```

Comment:
Maybe a little comment here to explain that it will fetch the number of cores used by rayon to divide the max-memory by it, and therefore this function must be called inside of a rayon thread pool?